lib/search_param_parser.rb

```diff
@@ -1,34 +1,71 @@
+    def parse_movie_params(params)
+      output = {
+        actor: params[:actor],
+        actor2: params[:actor2],
+        year: nil,
+        exact_year: nil,
+        year_select: params[:year_select],
+        before_year: nil,
+        after_year: nil,
+        genre: params[:genre],
+        company: params[:company],
+        mpaa_rating: params[:mpaa_rating],
+        page: (params[:page].presence || 1),
+        sort_by: (params[:sort_by].presence || 'revenue')
+      }
 
-  def parse_params(params)
-    @passed_params = params
+      year = params[:date][:year] if params[:date].present?
+      output[:year] = year
+      if year.present?
+        if params[:year_select].blank? || params[:year_select] == 'exact'
+          output[:exact_year] = year
+        end
+        output[:before_year] = "#{year}-01-01" if params[:year_select] == 'before'
+        output[:after_year] = "#{year}-12-31" if params[:year_select] == 'after'
+      end
+      output.select { |_k, v| v.present? }
+    end
 
-    @year_select = @passed_params[:year_select]
+    def parse_movie_params_for_display(params)
+      output = {}
+      output[:actor_display] = "#{params[:actor].titlecase} movies" if params[:actor].present?
+      output[:rating_display] = "Rated #{params[:mpaa_rating]}" if params[:mpaa_rating].present?
 
-    @year = @passed_params[:year]
+      if params[:genre].present?
+        genres = Movie::GENRES.to_h
+        genre_selected = genres.key(params[:genre].to_i)
+        output[:genre_display] = "#{genre_selected} movies"
+      end
 
-    if @year.present?
-      if @year_select.present?
-        if @year_select == "exact"
-          @passed_params[:exact_year] = @year
-        elsif @year_select == "after"
-          @passed_params[:after_year] = "#{@year}-12-31"
-        elsif @year_select == "before"
-          @passed_params[:before_year] = "#{@year}-01-01"
+      year = params[:date][:year] if params[:date].present?
+      if year.present?
+        output[:year_display] = case params[:year_select]
+                                when 'before' then "before #{year}"
+                                when 'after' then "after #{year}"
+                                else
+                                  "from #{year}"
         end
-      else
-        @passed_params[:exact_year] = @year
       end
-    end
 
-    if !@passed_params[:page]
-      @passed_params[:page] = 1
-    end #pagination
+      if params[:sort_by].present?
+        sort_options = Movie::SORT_BY.to_h
+        sort_key = sort_options.key(params[:sort_by])
+        output[:sort_display] = "sorted by #{sort_key}"
+      end
 
-    if !@passed_params.keys.include?('sort_by')
-      @passed_params[:sort_by] = "revenue"
+      output.select { |_k, v| v.present? }
+            .values
+            .join(', ')
     end
 
-  end #parase params
+    private
 
-end #final
+    def use_exact_year?(year_select)
+      year_select.blank? || year_select == 'exact'
+    end
```

Comment:
🤔 This doesn't seem to be used anywhere

Reply:
whoops! you're right. it was just part of the process and i forgot to clean it up.